Monika After Story/game/definitions.rpy

```diff
@@ -4594,7 +4594,23 @@ init -1 python:
         return appIds
 
 init 2 python:
-    # global functions that should be defined after level 0a
+    #Global functions that should be defined after level 0
+    import re
+    def regexInStr(teststr, keyword_list):
+        """
+        Checks if anything from keyword_list is in teststr checking using regex
+        IN:
+            teststr - string to check
+            keyword_list - list of words/prases to see if in teststr
+        OUT:
+            boolean:
+                - True if anything from keyword_list is in teststr
+                - False otherwise
+        NOTE: ignores case by default
+        """
+        re_compiled_list = re.compile('|'.join(keyword_list), re.IGNORECASE)
+        return bool(re_compiled_list.search(teststr))
```

Comment:
since we're not actually using regex here, doing a `return teststr in "|".join(keyword_list)` is faster than this

Reply:
disregard that, we are actually doing a regex here.

that being said, re.compile is not meant to be called everytime you want to search something. It should be called once, then mulitiple searches can be done with the compiled expression.

`regexInStr` as a function encourages use widespread use, which is not recommended practice, so I would revert this back to how it was before.